eng/versioning/pom_file_version_scanner.ps1

```diff
@@ -0,0 +1,370 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Use case: This script verifies the following:
+# 1. There are no duplicate entries in any of the version_*.txt files
+# 2. There are no duplicate entries in the external_dependencies.txt file
+# 3. POM file verification across the enlistment which includes the following:
+#    a. There are no <dependencyManagement> sections
+#    a. Every <dependency> and <plugin> has a <groupId>, <artifactId> and <version>
+#    b. Every <version> has the appropriate x-version-update tag
+#    c. The <version>'s value is the same as the value in the version_*txt file or external_dependency
+#
+# Output:
+# This script will process the entire enlistment. If any errors are encountered, will report them at
+# time they are encountered and continue processing.
+# Errors for any duplicate entries in the various version files will be reported before any POM files
+# are processed.
+# Errors for a given pom file will appear after the "processing pomFile=<fullPathAndFileName" line for
+# that file and before the processing line for the next file. Missing version x-version-update tag
+# errors will cause the output of what the tag should be in the cases where it is known (ie. pom file's
+# version or parent's version). In the cases of dependencies it'll output a tag and the appropriate
+# type (current, dependency or external_dependency) will have to be selected. For example:
+# <!-- {x-version-update;<groupId>:<artifactId>;current|dependency|external_dependency<select one>} -->"
+#
+# This script can be run locally from the root of an enlisment. .\eng\versioning\pom_file_version_scanner.ps1
+
+# Since we're skipping Management for the moment, only look for files with certain parents. These
+# limitations will vanish once Management track is updated.
+$ValidParents = ("azure-client-sdk-parent", "azure-data-sdk-parent", "azure-cosmos-parent")
+$Path = Resolve-Path ($PSScriptRoot + "/../../")
+
+# Not all POM files have a parent entry
+$PomFilesIgnoreParent = ("$($Path)\parent\pom.xml")
+$script:FoundError = $false
+$StartTime = $(get-date)
+
+class Dependency {
+    [string]$id
+    [string]$depVer
+    [string]$curVer
+    Dependency(
+        [string]$inputString
+    ){
+        $split = $inputString.Split(";")
+        $this.id = $split[0]
+        $this.depVer = $split[1]
+        $this.curVer = $split[2]
```

Comment:
Should this do any validation on the number of substrings? What happens if there is an improper amount of substrings?

Should this do a pattern match on the depVer or curVer to ensure they are proper Maven dependencies?

Reply:
@alzimmermsft it's trivial to add verification for the number of substrings and I can add that for correctness (though malformed entries would, in theory, get hit when the update_versions script is run). 

I really don't want to add anything that verifies everything is being used just yet only because Management is still being excluded until its centralized versioning work is done.